Sources/SwiftBSON/ExtendedJSONDecoder.swift

```diff
@@ -65,6 +60,38 @@ public class ExtendedJSONDecoder {
         return try bsonDecoder.decode(T.self, fromBSON: bson)
     }
 
+    /// Decodes an instance of the requested type `T` from the provided extended JSON data.
+    /// - SeeAlso: https://docs.mongodb.com/manual/reference/mongodb-extended-json/
+    ///
+    /// - Parameters:
+    ///   - type: Codable type to decode the input into.
+    ///   - data: `Data` which represents the JSON that will be decoded.
+    /// - Returns: Decoded representation of the JSON input as an instance of `T`.
+    /// - Throws: `DecodingError` if the JSON data is corrupt or if any value throws an error during decoding.
+    public func decode<T: Decodable>(_: T.Type, from data: Data) throws -> T {
+        try self.decodeBytes(T.self, from: data)
+    }
+
+    /// Decodes an instance of the requested type `T` from the provided extended JSON data.
+    /// - SeeAlso: https://docs.mongodb.com/manual/reference/mongodb-extended-json/
+    ///
+    /// - Parameters:
+    ///   - type: Codable type to decode the input into.
+    ///   - buffer: `ByteBuffer` which contains the JSON data that will be decoded.
+    /// - Returns: Decoded representation of the JSON input as an instance of `T`.
+    /// - Throws: `DecodingError` if the JSON data is corrupt or if any value throws an error during decoding.
+    public func decode<T: Decodable>(_: T.Type, from buffer: ByteBuffer) throws -> T {
+        guard buffer.readableBytes > 0 else {
+            throw DecodingError._extendedJSONError(keyPath: [], debugDescription: "empty buffer provided to decode")
+        }
+
+        var buffer = buffer
+        // readBytes never returns nil here because we checked that the buffer wasn't empty and only read
+        // readable bytes out from it.
+        // swiftlint:disable:next force_unwrapping
+        return try self.decodeBytes(T.self, from: buffer.readBytes(length: buffer.readableBytes)!)
```

Comment:
according to the `swift-extras-json` [readme](https://github.com/swift-extras/swift-extras-json#use-with-swiftnio-bytebuffer), it's more performant to read the bytes out of the buffer than it is to use `ByteBufferView` via `readableBytesView`. I think this is because the view relies on `readInteger` for each byte, which can be expensive to call repeatedly.